src/weather.nim

```diff
@@ -90,57 +54,50 @@ if paramCount() > 0:
     showHelp()
 
 # get settings if exist - if not create them wiht default version...
-while not getSettings():
+while not weather.getSettings():
   createDefaultSettings()
 
 # create combined latitude and longitude into one variable for Urls:
-let latlong = fmt"{Wthr.latConfig},{Wthr.lonConfig}"
+let latlong = fmt"{weather.latConfig},{weather.lonConfig}"
 
 # contruct DarkSky URL from settings:
-var darkSkyUrlFin: string
-darkSkyUrlFin = fmt"{Wthr.darkskyUrl}{Wthr.darkskyKey}/"
-darkSkyUrlFin.add(fmt"{latlong}?units=")
-darkSkyUrlFin.add(fmt"{Wthr.placeUnits}&exclude={Wthr.darkskyExclude}")
-when not defined(release):
-  echo "DEBUG: final DarkSky URL:\n" & darkSkyUrlFin & "\n"
-
-let rawWeatherData = returnWebSiteData(darkSkyUrlFin, Wthr)
+var darkSkyUrlFin = fmt"{weather.darkskyUrl}{weather.darkskyKey}/"
+darkSkyUrlFin &= fmt"{latlong}?units="
+darkSkyUrlFin &= fmt"{weather.placeUnits}&exclude={weather.darkskyExclude}"
+debug fmt"DEBUG: final DarkSky URL:\n{darkSkyUrlFin}\n"
+
+let rawWeatherData = returnWebSiteData(darkSkyUrlFin, weather)
 let weatherJson = returnParsedJson(rawWeatherData)
-extractWeather(weatherJson)
+weather.extractWeather(weatherJson)
 
 # Obtain Geo Location data
 # Provide a Google API from env variable 'GAPI' of set:
-if getEnv("GAPI").len > 0:
-  Wthr.googleKey = getEnv("GAPI")
+weather.googleKey = getEnv("GAPI", "")
```

Comment:
If you check ``getEnv`` documentation, you'll see that if the env variable doesn't exist it just returns an empty string. I explicitly added the second argument so it's easier to understand.